scripts/new-upstream-snapshot

```diff
@@ -243,9 +243,14 @@ main() {
         fail "failed to read Source from changelog"
     dist=$(dpkg-parsechangelog --show-field Distribution)
 
-    # if present, pull the '~16.04.x' off of the previous entry.
-    sru_ver_suff=$(echo "$prev_pkg_ver" |
-        sed 's,.*~\([0-9.]*\)[.][0-9]$,~\1.1,')
+    sru_ver_suff=$(distro-info --series ${cur_branch#ubuntu/} --release |
+                       awk '{printf("~%s.1", $1)}') ||
```

Comment:
Why are we hard coding a .1 here? Won't there possibly be a .2 and such down the road?

Reply:
@mitechie, right new-upstream-snapshot will generally be run on our first major import from master of a new <major>-<minor>-<gitcommit> version into a given Ubuntu release. So we can, and do always hardcode `YY.MM.1`  as the first upload into 16.04 as ~16.04.1 for cloud-init.

`new-upstream-snapshot` also does do that increment you mentioned from .1 to ,2 when adding new changelog entries in two other scenarios:
   1. refreshing existing quilt patches in debian/patches "cherry-picks" to sync with upstream commits that affect the quilt patch
   2. dropping quilt patches whose upstream commit is now merged into our ubuntu/* branch

Any further quilt patches or release specific changes required after that fact would result in manual incrementing that sub-revision to .2 or .3  as `new-upstream-snapshot` doesn't try to automate  those cases